rdr_service/dao/genomics_dao.py

```diff
@@ -580,13 +580,18 @@ def get_gem_consent_removal_date(self, member):
         # Calculate gem consent removal date
         # Earliest date between GROR or Primary if both,
         withdraw_dates = []
-        if consent_status.consentForGenomicsROR != QuestionnaireStatus.SUBMITTED:
+        if consent_status.consentForGenomicsROR != QuestionnaireStatus.SUBMITTED and \
+            consent_status.consentForGenomicsRORAuthored:
             withdraw_dates.append(consent_status.consentForGenomicsRORAuthored)
 
-        if consent_status.consentForStudyEnrollment != QuestionnaireStatus.SUBMITTED:
+        if consent_status.consentForStudyEnrollment != QuestionnaireStatus.SUBMITTED and \
+            consent_status.consentForStudyEnrollmentAuthored:
```

Comment:
Is `consentForStudyEnrollment` ever anything but `SUBMITTED`? I think what we've been needing to do is remove the participant summary altogether if we need to remove their Primary consent status.

Reply:
I didn't realize that we remove the records. If that is the case, then this is really just a safety precaution. I will add a check for the withdrawal status as well.